### PR TITLE
gh #21 DS Video Device Interface Update

### PR DESCRIPTION
**Pull request by krishna-1809**

### CODE REVIEW

File: include/dsVideoDevice.h [hunk 1]
```diff
@@ -387,9 +401,12 @@ dsError_t dsSetDisplayframerate(intptr_t handle, char *framerate);
  *
  */
 typedef void (*dsRegisterFrameratePreChangeCB_t)(unsigned int tSecond);
+
 /**
  * @brief This function is used to register the callback function for the Display framerate pre change event.
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:408]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 2]
```diff
@@ -416,10 +434,13 @@ dsError_t dsRegisterFrameratePreChangeCB(dsRegisterFrameratePreChangeCB_t CBFunc
  *
  */
 typedef void (*dsRegisterFrameratePostChangeCB_t)(unsigned int tSecond);
+
 /**
  * @brief This function is used to register a callback function for the Display framerate 
  *                      post change event from the HAL side.
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:442]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 3]
```diff
@@ -358,6 +370,8 @@ dsError_t dsGetCurrentDisplayframerate(intptr_t handle, char *framerate);
 /**
  * @brief Sets the display framerate for the device
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:373]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 4]
```diff
@@ -333,6 +343,8 @@ dsError_t dsGetFRFMode(intptr_t handle, int *frfmode);
 /**
  * @brief Gets the current framerate of the device
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:346]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 5]
```diff
@@ -310,6 +318,8 @@ dsError_t dsSetFRFMode(intptr_t handle, int frfmode);
 /**
  * @brief Gets the FRF mode of the device
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:321]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 6]
```diff
@@ -286,6 +292,8 @@ dsError_t dsForceDisableHDRSupport(intptr_t handle, bool disable);
 /**
  * @brief Sets the FRF mode of the device
  *
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:295]: Confirmed with Ranjith(Frame Rate dev.) that it is specific to TVs. So all the Frame Rate related API are sink specific

File: include/dsVideoDevice.h [hunk 7]
```diff
@@ -243,6 +247,8 @@ dsError_t dsGetSupportedVideoCodingFormats(intptr_t handle, unsigned int * suppo
 /**
  * @brief Gets the video codec information
  *
+ * For sink devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
+ *
```
Comment (srinivasgtl) [inline include/dsVideoDevice.h:251]: Please update this from source point of view as well.  The description should clearly defined on what is expected from this API.

File: include/dsVideoDevice.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (srinivasgtl) [reply include/dsVideoDevice.h:295]: Can we have some description for the sink devices here?

**Review by srinivasgtl — COMMENTED** [review]
Please update the descriptions with respect to source and sink devices.

File: include/dsVideoDevice.h [hunk 8]
```diff
@@ -131,6 +131,8 @@ dsError_t  dsGetVideoDevice(int index, intptr_t *handle);
 /**
  * @brief Sets the screen zoom mode (decoder format conversion)
  *
+ * For sink devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:135]: Updated

File: include/dsVideoDevice.h [hunk 9]
```diff
@@ -154,6 +156,8 @@ dsError_t  dsSetDFC(intptr_t handle, dsVideoZoom_t dfc);
 /**
  * @brief Gets the screen zoom mode (decoder format conversion)
  *
+ * For sink devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:161]: Updated

File: include/dsVideoDevice.h (hunk 7 above)
Comment (krishna-1809) [reply include/dsVideoDevice.h:254]: Updated

File: include/dsVideoDevice.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsVideoDevice.h:299]: Updated

File: include/dsVideoDevice.h (hunk 7 above)
Comment (srinivasgtl) [reply include/dsVideoDevice.h:254]: For the said API, the video codec is the input, while the description says it returns the "supported video codes"

Comment (srinivasgtl) [reply include/dsVideoDevice.h:254]: Can you come back on this whether we need this API or need to deprecate?

Comment (Srigayathry) [reply include/dsVideoDevice.h:254]: As per the middleware team input, we cannot remove it because  it is used for diagnostic purpose.

**Review by ddevadas53276 — APPROVED** [review]

File: include/dsVideoDevice.h (hunk 7 above)
Comment (Anbukannadhasan) [reply include/dsVideoDevice.h:254]: dsVideoCodecInfo_t  - Repharse the statement as below(This function not return supported video codecs)
"For source devices, this function returns the supported codec profile and levels for the input coding format."

**Review by hari22yuva — APPROVED** [review]
Approved

**Review by ddevad — APPROVED** [review]
ship it

**Pull request merged**